src/ol/renderer/layerrenderer.js

```diff
@@ -95,6 +97,33 @@ ol.renderer.Layer.prototype.handleImageChange = function(event) {
 
 
 /**
+ * Load the image if not already loaded, and manage the
+ * registration/deregistration of the image change listener.
+ * @param {ol.ImageBase} image Image.
+ * @return {boolean} `true` if the image is already loaded, `false`
+ *     otherwise.
+ * @protected
+ */
+ol.renderer.Layer.prototype.loadImage = function(image) {
+  var imageState = image.getState();
+  if (imageState == ol.ImageState.LOADED ||
+      imageState == ol.ImageState.ERROR) {
+    goog.events.unlisten(image, goog.events.EventType.CHANGE,
+        this.handleImageChange_, false, this);
+  } else {
+    if (imageState == ol.ImageState.IDLE) {
+      image.load();
+    }
```

Comment:
Right. http://stackoverflow.com/questions/12354865/image-onload-event-and-browser-cache also mentions that problem. I'll change the order: register the listener first, and then call load if the image is idle.